CONTRIBUTING.md

```diff
@@ -2,51 +2,59 @@
+- If you haven't already, complete the Contributor License Agreement ("CLA").
 
 ## Contributor License Agreement ("CLA")
 In order to accept your pull request, we need you to submit a CLA. You only need
-to do this once to work on any of Facebook's open source projects.
+to do this once to work on any of Facebook's open source projects. Complete your CLA [here](https://code.facebook.com/cla).
+
+## Versioning / release workflow
+We use [semantic versioning](https://semver.org/). To prepare a release:
+1. Update version numbers.
+2. Update the change log.
+3. Make sure all tests are passing.
+4. Create a release tag with change log summary using the github release interface (release tag should follow semantic versioning as described above)
 
-Complete your CLA here: <https://code.facebook.com/cla>
+Stable version is assigned by Habitat core team based on regularly tested versions
 
 ## Issues
-We use GitHub issues to track public bugs. Please ensure your description is
+We use [GitHub issues](../../issues) to track public bugs. Please ensure your description is
 clear and has sufficient instructions to be able to reproduce the issue.
 
-## Test
-Please make sure that C++ unit tests are passing and run the example script:
-```
-python ./example.py
-```
-
-## Check typing
-We use mypy to check Python typing and guard API consistency, please make sure next command doesn't complain prior to submission:
-```
-mypy . --ignore-missing-imports
-```
 
 ## Coding Style
-We use `clang-format` for linting and code style enforcement of c++ code.
-Code style follows the [Google C++ guidelines](https://google.github.io/styleguide/cppguide.html).
-Install `clang-format` through `brew install clang-format` on MacOS, or by downloading binaries or sources from http://releases.llvm.org/download.html for Ubuntu etc.
-For vim integration add to your .vimrc file `map <C-K> :%!clang-format<cr>` and use Ctrl+K to format entire file.
-Integration plugin for [vscode](https://marketplace.visualstudio.com/items?itemName=xaver.clang-format).
 
-We use `black` for linting and code of python code.
-Install `black` through `pip install black`
+- C++
+  - In general, we follow [C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines) and [Google C++ guidelines](https://google.github.io/styleguide/cppguide.html)
+  - Use `clang-format` for style enforcement and linting. Install clang-format through `brew install clang-format` on MacOS, or by downloading [binaries or sources](http://releases.llvm.org/download.html) for Ubuntu etc.
+- Python
+  - We follow PEP8 and use typing
+  - Use `black` for style enforcement and linting. Install black through pip install black. 
+  
+We also use pre-commit hooks to ensure linting and style enforcement. Install the pre-commit hooks with `pip install pre-commit && pre-commit install`.
+
+## Documentation
+- Our documentation style is based on Magnum / Corrade and uses a similar build system
```

Comment:
Add link to magnum's style guide. (There is one!)

Reply:
There are two, in fact -- [1](https://doc.magnum.graphics/corrade/corrade-coding-style.html), [2](https://doc.magnum.graphics/magnum/coding-style.html) :) But I'm not sure how compatible Magnum's and Habitat's coding style is (especially re shared pointer usage, file / variable naming etc.) -- so maybe mention that ~~it shouldn't be taken too literally~~ only the doc-related part is relevant.

And if links, then maybe https://mcss.mosra.cz/documentation/python/ and/or https://mcss.mosra.cz/documentation/doxygen/ as total full and complete references of what's all possible?